apps/retrospective/services/sprint.go

```diff
@@ -127,19 +127,129 @@ func (service SprintService) Get(sprintID string) (*retroSerializers.Sprint, int
 		return nil, http.StatusInternalServerError, errors.New("failed to get sprint")
 	}
 
+	summary, status, err := service.GetSprintSummary(sprintID, sprint.RetrospectiveID)
+	if err != nil {
+		return nil, status, err
+	}
+
+	sprint.Summary = summary
+
 	err = db.Model(&retroModels.SprintSyncStatus{}).
 		Where("sprint_id = ?", sprintID).
 		Order("created_at desc").
 		Select("status").
 		Row().Scan(&sprint.SyncStatus)
 
 	if err != nil {
+		if err != gorm.ErrRecordNotFound {
+			utils.LogToSentry(err)
+			return nil, http.StatusInternalServerError, errors.New("failed to get sprint")
+		}
 		sprint.SyncStatus = int8(retroModels.NotSynced)
 	}
 
 	return &sprint, http.StatusOK, nil
 }
 
+// GetSprintSummary ...
+func (service SprintService) GetSprintSummary(sprintID string, retroID uint) (summary retroSerializers.SprintSummary, status int, err error) {
+	db := service.DB
+
+	var sprint retroModels.Sprint
+
+	err = db.Model(&retroModels.Sprint{}).Where("id = ?", sprintID).Preload("Retrospective").First(&sprint).Error
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return summary, http.StatusNotFound, errors.New("sprint not found")
+		}
+		utils.LogToSentry(err)
+		return summary, http.StatusInternalServerError, errors.New("failed to get sprint summary")
+	}
+
+	err = db.Model(&retroModels.Sprint{}).
+		Scopes(retroModels.SprintJoinSM).
+		Where("sprints.id = ?", sprintID).
+		Select("COUNT(*) as member_count, " +
+			"SUM(allocation_percent) as total_allocation, " +
+			"SUM(expectation_percent) as total_expectation, " +
+			"sum(vacations) as total_vacations," +
+			"0 as holidays").
+		Scan(&summary).Error
+
+	if err != nil {
+		utils.LogToSentry(err)
+		return summary, http.StatusInternalServerError, errors.New("failed to get sprint summary")
```

Comment:
shouldn't we return nil instead of summary

Reply:
Done